File: src/main/java/uk/co/ben_gibson/git/link/UI/LineSelection.java
package uk.co.ben_gibson.git.link.UI;

public class LineSelection
{
    private int start;
    private int end;


    public LineSelection(final int start)
    {
        this.start = start;
        this.end   = start;
    }


    public LineSelection(final int start, final int end)
    {
        this.start = start;
        this.end   = end;
    }


    public int start()
    {
        return start;
    }


    public int end()
    {
        return end;
    }


    public boolean isSingleLineSelection()
    {
        return (start == end);
    }


    public boolean isMultiLineSelection()
    {
        return !isSingleLineSelection();
    }
}

File: src/main/kotlin/uk/co/ben_gibson/git/link/ui/LineSelection.kt
package uk.co.ben_gibson.git.link.ui

data class LineSelection(val start: Int, val end: Int) {
    init {
        require(start > 0) { "Start line must be positive, got $start" }
        require(end > 0) { "End line must be positive, got $end" }
        require(start <= end) { "Start line ($start) must be <= end line ($end)" }
    }

    constructor(line: Int) : this(line, line)

    val lineCount: Int
        get() = end - start + 1
}